src/offset_vector.c

```diff
@@ -66,19 +66,28 @@ uint32_t _aoi_Next(void *ctx, RSQueryTerm **term);
 void _aoi_Free(void *ctx);
 void _aoi_Rewind(void *ctx);
 
-void *_newAggregateIter() {
+static void *aggiterNew() {
   _RSAggregateOffsetIterator *it = malloc(sizeof(_RSAggregateOffsetIterator));
   it->size = 0;
   it->offsets = NULL;
   it->iters = NULL;
   it->terms = NULL;
   return it;
 }
+
+static void aggiterFree(void *p) {
+  _RSAggregateOffsetIterator *aggiter = p;
+  free(aggiter->offsets);
```

Comment:
Those are initialize to NULL, when we reach here they are surly not NULL?

Reply:
free(NULL) is valid